src/vcpkg/archives.cpp

```diff
@@ -6,166 +6,178 @@
         {
-            static bool recursion_limiter_sevenzip = false;
-            Checks::check_exit(VCPKG_LINE_INFO, !recursion_limiter_sevenzip);
-            recursion_limiter_sevenzip = true;
-            const auto seven_zip = paths.get_tool_exe(Tools::SEVEN_ZIP);
-            const auto code_and_output =
-                cmd_execute_and_capture_output(Command{seven_zip}
-                                                   .string_arg("x")
-                                                   .path_arg(archive)
-                                                   .string_arg(Strings::format("-o%s", to_path_partial))
-                                                   .string_arg("-y"));
-            Checks::check_exit(VCPKG_LINE_INFO,
-                               code_and_output.exit_code == 0,
-                               "7zip failed while extracting '%s' with message:\n%s",
-                               archive,
-                               code_and_output.output);
-            recursion_limiter_sevenzip = false;
+            win32_extract_with_seven_zip(paths, archive, to_path);
         }
 #else
-        if (ext == ".gz" || ext == ".bz2")
+        if (ext == ".zip")
         {
             const auto code =
-                cmd_execute(Command{"tar"}.string_arg("xzf").path_arg(archive), InWorkingDirectory{to_path_partial});
-            Checks::check_exit(VCPKG_LINE_INFO, code == 0, "tar failed while extracting %s", archive);
+                cmd_execute(Command{"unzip"}.string_arg("-qqo").path_arg(archive), InWorkingDirectory{to_path});
+            Checks::check_exit(VCPKG_LINE_INFO, code == 0, "unzip failed while extracting %s", archive);
         }
-        else if (ext == ".zip")
+#endif
+        else if (ext == ".gz" || ext == ".bz2" || ext == ".tgz")
         {
-            const auto code =
-                cmd_execute(Command{"unzip"}.string_arg("-qqo").path_arg(archive), InWorkingDirectory{to_path_partial});
-            Checks::check_exit(VCPKG_LINE_INFO, code == 0, "unzip failed while extracting %s", archive);
+            vcpkg::extract_tar(paths.get_tool_exe(Tools::TAR), archive, to_path);
         }
         else
         {
             Checks::exit_maybe_upgrade(VCPKG_LINE_INFO, "Unexpected archive extension: %s", ext);
         }
-#endif
-
-        fs.rename(to_path_partial, to_path, ec);
+    }
+}
 
-        using namespace std::chrono_literals;
+namespace vcpkg
+{
+    void extract_tar(const Path& tar_tool, const Path& archive, const Path& to_path)
+    {
+        const auto code =
+            cmd_execute(Command{tar_tool}.string_arg("xzf").path_arg(archive), InWorkingDirectory{to_path});
+        Checks::check_exit(VCPKG_LINE_INFO, code == 0, "tar failed while extracting %s", archive);
+    }
 
-        auto retry_delay = 8ms;
+    Path extract_archive_to_temp_subdirectory(const VcpkgPaths& paths, const Path& archive, const Path& to_path)
```

Comment:
>(argument to_path is used only to create a partial name -- why not just pass in the partial name?)

It's also used to form the return value.

>Since it's only used in this file anyway, I'd feel better if it was static.

Ah, this extraction was leftover from when I was trying to avoid using npm; it can indeed be static now.

Reply:
It is static (by virtue of being in an unnamed namespace)